### ISSUE TITLE
关于batch_norm layer的参数自命名

### ISSUE THREAD

**Issue by hiahiahu** [PaddlePaddle/Paddle#952]
因为要共享参数，对batch_norm layer，配置如下：
Layer(name=name + 'batch_norm1',
          type='batch_norm',
          active_type="relu",
          use_global_stats=True,
          bias=Bias(parameter_name = parse_parameter(name) +'BN1.bias',
          initial_std=0.02, is_static=static, learning_rate=lr),
          inputs=Input(name + 'conv1',
                       parameter_name = parse_parameter(name) +'BN1.w',
                       initial_std=0.02,
                       is_static=static,
                       learning_rate=lr,
                       image=Image(channels=size, img_size=1)), )

训练日志中
I1219 22:16:30.714959 18400 TrainerInternal.cpp:207] title_block_2_t3_BN1.w avg_abs_val=0.0155404   max_val=0.0588142   avg_abs_grad=0.103073    max_grad=0.509655
I1219 22:16:30.715075 18400 TrainerInternal.cpp:207] _title_block_2_t3_batch_norm1.w1 avg_abs_val=0.00608151  max_val=0.0277185   avg_abs_grad=0           max_grad=0
I1219 22:16:30.715212 18400 TrainerInternal.cpp:207] _title_block_2_t3_batch_norm1.w2 avg_abs_val=3.72902e-05 max_val=0.00016382  avg_abs_grad=0           max_grad=0
I1219 22:16:30.715330 18400 TrainerInternal.cpp:207] title_block_2_t3_BN1.bias avg_abs_val=0.017024    max_val=0.0620825   avg_abs_grad=0.0651094   max_grad=0.299386

其中两个 _title 开头的参数 怎么自己设置命名？ 好像没有lstm里面的para_prefix 这么个参数？
看config_parse 这两个layer似乎是在这里添加进去的
use_gpu = bool(int(g_command_config_args.get("use_gpu", 0)))
        is_shared = True if not use_gpu else False
        for i in xrange(2):
            inputs.append(Input(inputs[0].input_layer_name,
                                initial_std=0.0,
                                initial_mean=0.0,
                                is_static=True,
                                 is_shared=is_shared,
                                ))

这里设置的 is_static， 为什么日志文件中的average_val和max_val不是0 ？
求解答！辛苦~

**Comment by qingqing01** [comment]
> 其中两个 _title 开头的参数 怎么自己设置命名？

参数 ` _title_block_2_t3_batch_norm1.w1` 和 `_title_block_2_t3_batch_norm1.w2 ` 确实是 `config_parse .py`解析时自动添加的；名字依据 该层layer的名字 即  `name + 'batch_norm1`  自动产生的 。

>  这里设置的 is_static， 为什么日志文件中的average_val和max_val不是0 ？

设置 `is_static = True`是对 参数  ` _title_block_2_t3_batch_norm1.w1` 和 `_title_block_2_t3_batch_norm1.w2 ` 起作用， 表示改参数不需要梯度、不需要更新梯度信息。 从log里可以看出grad为0： `avg_abs_grad=0 max_grad=0` ， 但是本身这两moving average参数得值不为0。

**Comment by hiahiahu** [comment]
_title_block_2_t3_batch_norm1.w1 和 _title_block_2_t3_batch_norm1.w2 两个参数有没有办法自己指定命名(改config_parse？。。) 因为是ltr任务，好像没法容忍参数不一致，即使该参数不更新，也要保持一致

**Comment by qingqing01** [comment]
@hiahiahu 我正在修改，加一个prefix，可以允许用户自定义~

**Comment by hiahiahu** [comment]
辛苦 谢谢！

**Comment by qingqing01** [comment]
@hiahiahu   我还有个疑问，BatchNorm用在 LTR任务中，如果两个Layer的feature map本身就不一样的话， moving average的参数可以共享吗？

**Comment by lcy-seso** [comment]
即使前一层的参数共享，Batch Norm 的参数不应该被共享，才更合理。

**Comment by qingqing01** [comment]
@lcy-seso  如果不合理，就不加了~

**Comment by hiahiahu** [comment]
恩  似乎是我理解有问题 我看下具体参数说明先~

**Comment by lcy-seso** [comment]
现在没有被共享的两个参数是 mean 和  std 的 moving average，即使浅层经过某个参数共享的layer ，但是 batch norm 的输入肯定是不一样的。mean  和 std 不能共享的话，那这一层本身的两个参数 ，论文里面的 gamma 和 beta 再被共享也不是非常合理了。我觉得 batch norm 这一层的参数（mean / std / gamma / beta ）都不应该被共享。
